java/spi.java.hints/src/org/netbeans/modules/java/hints/spiimpl/Utilities.java

```diff
@@ -1352,6 +1354,33 @@ protected JCModifiers modifiersOpt(JCModifiers partial) {
             return super.modifiersOpt(partial);
         }
 
+
+        public JCVariableDecl formalParameter(boolean lambdaParam, boolean recordComponents) {
+            if (token.kind == TokenKind.IDENTIFIER) {
+                if (token.name().startsWith(dollar)) {
+                    com.sun.tools.javac.util.Name name = token.name();
+
+                    Token peeked = S.token(1);
+
+                    if (peeked.kind == TokenKind.COMMA || peeked.kind == TokenKind.RPAREN) {
+                        nextToken();
+                        return JackpotTrees.createVariableWildcard(ctx, name);
+                    }
+                }
+            }
+                        JCTree.JCVariableDecl result = null;
+            try {
+                Class[] paramTypes = {boolean.class, boolean.class};
+                result = (JCTree.JCVariableDecl) MethodHandles.lookup()
+                        .findSpecial(JavacParser.class, "formalParameter", MethodType.methodType(JCTree.JCVariableDecl.class, paramTypes), JackpotJavacParser.class) // NOI18N
+                        .invoke(this, lambdaParam, recordComponents);
+            } catch (Throwable e) {
+                Logger.getLogger(NBParserFactory.class.getName()).log(Level.FINE, null, e);
```

Comment:
I think it should be fine to just fail here, shouldn't it (the method must exist, as the current method overrides it):
```suggestion
                throw new IllegalStateException(ex);
```